internal/cmd/dev.go

```diff
@@ -89,16 +98,390 @@ func (s *statement) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
-func errorJson(e error) []errorResponse {
-	return []errorResponse{
-		{
-			Error: struct {
-				Message string `json:"message"`
-			}{
-				Message: e.Error(),
-			},
-		},
+func errorJson(e error) *hranaError {
+	return hranaErr(e)
+}
+
+var errSqliteUnknown = "SQLITE_UNKNOWN"
+
+type hranaError struct {
+	Message string  `json:"message"`
+	Code    *string `json:"code,omitempty"`
+}
+
+func hranaErr(e error) *hranaError {
+	err := new(hranaError)
+	err.Message = e.Error()
+	err.Code = &errSqliteUnknown
+	return err
+}
+
+type hranaCol struct {
+	Name string `json:"name"`
+}
+
+type hranaValue struct {
+	Type   string      `json:"type"`
+	Value  interface{} `json:"value,omitempty"`
+	Base64 string      `json:"base64,omitempty"`
+}
+
+type hranaNamedArg struct {
+	Name  string     `json:"name"`
+	Value hranaValue `json:"value"`
+}
+
+type hranaStmt struct {
+	Sql       *string          `json:"sql,omitempty"`
+	SqlId     *int32           `json:"sql_id,omitempty"`
+	Args      *[]hranaValue    `json:"args,omitempty"`
+	NamedArgs *[]hranaNamedArg `json:"named_args,omitempty"`
+	WantRows  *bool            `json:"want_rows"`
+}
+
+type hranaCond struct {
+	Type  string       `json:"type"`
+	Step  *int         `json:"step,omitempty"`
+	Cond  *hranaCond   `json:"cond,omitempty"`
+	Conds *[]hranaCond `json:"conds,omitempty"`
+}
+
+type hranaBatchStep struct {
+	Stmt *hranaStmt `json:"stmt,omitempty"`
+	Cond *hranaCond `json:"condition,omitempty"`
+}
+
+type hranaBatch struct {
+	Steps []hranaBatchStep `json:"steps"`
+}
+
+type hranaExecuteReq struct {
+	Stmt hranaStmt `json:"stmt"`
+}
+
+type hranaBatchReq struct {
+	Batch hranaBatch `json:"batch"`
+}
+
+type hranaStmtResult struct {
+	Cols     []hranaCol     `json:"cols"`
+	Rows     [][]hranaValue `json:"rows"`
+	Affected int32          `json:"affected_row_count"`
+	LastIR   *string        `json:"last_insert_rowid"`
+}
+
+type hranaSuccessResponse struct {
+	Result hranaStmtResult `json:"result"`
+}
+
+type hranaBatchResponse struct {
+	Results []*hranaStmtResult `json:"step_results"`
+	Errors  []*hranaError      `json:"step_errors"`
+}
+
+type hranaBatchSuccessResponse struct {
+	Result hranaBatchResponse `json:"result"`
+}
+
+func convertValue(val hranaValue) (interface{}, error) {
+	if val.Type == "blob" {
+		return []byte(val.Base64), nil
+	} else if val.Type == "integer" {
+		return strconv.Atoi(val.Value.(string))
```

Comment:
Hrana (and SQLite) integers are 64-bit, but the Go `int` may be 32-bit, so we should use `strconv.ParseInt()` instead of `Atoi()`.